src/registrar/admin.py

```diff
@@ -175,31 +175,57 @@ class DomainAdmin(ListHeaderAdmin):
     readonly_fields = ["state"]
 
     def response_change(self, request, obj):
-        PLACE_HOLD = "_place_client_hold"
-        EDIT_DOMAIN = "_edit_domain"
-        if PLACE_HOLD in request.POST:
-            try:
-                obj.place_client_hold()
-            except Exception as err:
-                self.message_user(request, err, messages.ERROR)
-            else:
-                self.message_user(
-                    request,
-                    (
-                        "%s is in client hold. This domain is no longer accessible on"
-                        " the public internet."
-                    )
-                    % obj.name,
-                )
-            return HttpResponseRedirect(".")
-        elif EDIT_DOMAIN in request.POST:
-            # We want to know, globally, when an edit action occurs
-            request.session["analyst_action"] = "edit"
-            # Restricts this action to this domain (pk) only
-            request.session["analyst_action_location"] = obj.id
-            return HttpResponseRedirect(reverse("domain", args=(obj.id,)))
+        ACTION_BUTTONS = {
+            "PLACE_HOLD": "_place_client_hold",
+            "REMOVE_HOLD": "_remove_client_hold",
+            "EDIT_DOMAIN": "_edit_domain",
+        }
+        if ACTION_BUTTONS["PLACE_HOLD"] in request.POST:
+            return self.do_place_client_hold(request, obj)
+        elif ACTION_BUTTONS["REMOVE_HOLD"] in request.POST:
+            return self.do_remove_client_hold(request, obj)
+        elif ACTION_BUTTONS["EDIT_DOMAIN"] in request.POST:
+            return self.do_edit_domain(request, obj)
```

Comment:
(Optional but Nice to have) Theres a neat [trick](https://stackoverflow.com/questions/9168340/using-a-dictionary-to-select-function-to-execute) you can do in python where you can actually embed a function as an 'object' inside of a dictionary, as so:
`myDict = {
    "PLACE_HOLD": self.do_place_client_hold,
    "REMOVE_HOLD": self.do_remove_client_hold,
    ...
}
`
Then you can call it:
`myDict[name]()` or `myDict.get(name, lambda: 'Invalid')()`

Basically I think you should be able to do something akin to this, in sorta psuedocode:
```
if action is in request.post:
    return action_buttons.get(request.post (maybe?))()
```

This might be worth exploring. I think in one of Alysias draft PRS shes adding quite a few more buttons to this area so, since we will need to refactor either way, storing all that inside of

Reply:
@zandercymatics thank you for this suggestion.  i have refactored response_change to incorporate your comment